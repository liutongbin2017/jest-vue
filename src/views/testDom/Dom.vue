<template>
  <div>
    <test-dom :message="messages" />
  </div>
</template>

<script>
import testDom from "@/views/testDom/TestDom";

export default {
  name: "Dom",
  data() {
    return {
      messages: ["Hello Jest", "Hello Vue"],
    };
  },
  components: {
    testDom,
  },
};
</script>
