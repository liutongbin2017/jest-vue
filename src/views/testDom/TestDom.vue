<template>
  <div class="outer">
    <p v-for="message in message" :key="message">{{ message }}</p>
    <!-- <span v-if="isAdmin">Admin Privileges</span>
    <span v-else>Not Authorized</span> -->

    <button>{{ msg }}</button>
  </div>
</template>

<script>
export default {
  // props: ["messages"],
  props: {
    message: {
      type: Array,
      required: true,
    },
    // msg: {
    //   type: String,
    //   required: true,
    // },
  },
  data() {
    return {};
  },
};
</script>
