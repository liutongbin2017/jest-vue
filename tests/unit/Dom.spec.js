import { mount, shallowMount, shallow } from "@vue/test-utils";
import Dom from "@/views/testDom/Dom.vue";
import TestDom from "@/views/testDom/TestDom.vue";

import flushPromises from "flush-promises";

describe("Dom.vue", () => {
  let wrapper, vm;
  // 测试代码运行前的操作
  beforeEach(() => {
    const message = ["Cat"];
    wrapper = mount(Dom);
    vm = wrapper.vm;
    wrapper.setData({ messages: message });
  });
  //传参测试，按照官网没反应
  it('equals messages to ["Cat"]', async () => {
    // await wrapper.setProps({ messages: ["Cat"] });
    // await flushPromises();
    console.log(wrapper.html(), vm.messages);
    expect(vm.messages).toEqual(["Cat"]);
  });
  // 为App的单元测试增加快照（snapshot）：比较dom差异
  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
  // Dom渲染测试
  // it("Dom组件渲染是否完成", () => {
  //   // 使用Vue组件选择器
  //   expect(wrapper.is(TestDom)).toBe(true);
  //   // 使用CSS选择器
  //   expect(wrapper.is(".outer")).toBe(true);
  //   // 使用CSS选择器
  //   expect(wrapper.contains("p")).toBe(true);
  // });
  // props：返回 Wrapper vm 的 props 对象。

  // it("接收到了bye作为Props", () => {
  //   wrapper = shallowMount(TestDom, {
  //     propsData: {
  //       messages: ["bye"],
  //     },
  //   });
  //   expect(wrapper.props("messages")).toContain("bye");
  // });
  // describe("SubmitButton.vue", () => {
  //   it("displays a non authorized message", () => {
  //     const msg = "submit";
  //     const message = ["bye", "Hello Jest"];
  //     const wrapper = shallowMount(TestDom, {
  //       propsData: {
  //         msg: msg,
  //         message: message,
  //       },
  //     });

  //     console.log(wrapper.html());
  //     expect(wrapper.find("span").text()).toBe("Not Authorized");
  //     expect(wrapper.find("button").text()).toBe("submit");
  //     expect(wrapper.props("message")).toContain("bye");
  //   });
  // });
});
