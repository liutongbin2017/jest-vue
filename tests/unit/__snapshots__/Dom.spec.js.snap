// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dom.vue has the expected html structure 1`] = `
<div>
  <div
    class="outer"
  >
    <p>
      Cat
    </p>
     
    <button>
      
    </button>
  </div>
</div>
`;
